creation/web_base/glidein_startup.sh

```diff
@@ -152,1932 +76,533 @@ do_start_all() {
-echo
-if [ ${ret} -ne 0 ]; then
-    warn "Error running '${last_script}'"
-fi
+    done
 
-#Things like periodic scripts might put messages here if they want them printed in the (stderr) logfile
-echo "=== Exit messages left by periodic scripts ===" 1>&2
-if [ -f exit_message ]; then
-    cat exit_message 1>&2
-else
-    echo "No message left" 1>&2
+    #############################
+    fixup_condor_dir
+
+    ##############################
+    # Start the glidein main script
+    add_config_line "GLIDEIN_INITIALIZED" "1"
+    log_write "glidein_startup.sh" "text" "Starting the glidein main script" "info"
+    log_write "glidein_startup.sh" "file" "${glidein_config}" "debug"
+    send_logs_to_remote          # checkpoint
+    echo "# --- Last Script values ---" >> glidein_config
+    last_startup_time=$(date +%s)
+    let validation_time=${last_startup_time}-${startup_time}
+    echo "=== Last script starting $(date) (${last_startup_time}) after validating for ${validation_time} ==="
+    echo
+    ON_DIE=0
+    trap 'signal_ignore' SIGHUP
+    signal_trap_with_arg 'signal_on_die' SIGTERM SIGINT SIGQUIT
+    #trap 'signal_on_die' TERM
+    #trap 'signal_on_die' INT
+    gs_id_work_dir=$(get_work_dir main)
+    "${main_dir}"/error_augment.sh -init
+    "${gs_id_work_dir}/${last_script}" glidein_config &
+    wait $!
+    ret=$?
+    if [ ${ON_DIE} -eq 1 ]; then
+        ret=0
+    fi
+    last_startup_end_time=$(date +%s)
+    "${main_dir}"/error_augment.sh  -process ${ret} "${last_script}" "${PWD}" "${gs_id_work_dir}/${last_script} glidein_config" "${last_startup_time}" "${last_startup_end_time}"
+    "${main_dir}"/error_augment.sh -concat
+    let last_script_time=${last_startup_end_time}-${last_startup_time}
+    echo "=== Last script ended $(date) (${last_startup_end_time}) with code ${ret} after ${last_script_time} ==="
+    echo
+    if [ ${ret} -ne 0 ]; then
+        log_warn "Error running '${last_script}'"
+    fi
+
+    #############################
+    # Things like periodic scripts might put messages here if they want them printed in the (stderr) logfile
+    echo "=== Exit messages left by periodic scripts ===" 1>&2
+    if [ -f exit_message ]; then
+        cat exit_message 1>&2
+    else
+        echo "No message left" 1>&2
+    fi
+    echo 1>&2
+
+    #########################
+    # clean up after I finish
+    glidein_exit ${ret}
+}
+
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
+    # Include all source scripts
+    source "$GWMS_SOURCEDIR"/utils_gs_http.sh
+    source "$GWMS_SOURCEDIR"/utils_gs_filesystem.sh
+    source "$GWMS_SOURCEDIR"/utils_gs_log.sh
+    source "$GWMS_SOURCEDIR"/utils_gs_signals.sh
+    source "$GWMS_SOURCEDIR"/utils_gs_tarballs.sh
+    source "$GWMS_SOURCEDIR"/utils_log.sh
+    source "$GWMS_SOURCEDIR"/utils_params.sh
+    source "$GWMS_SOURCEDIR"/utils_signals.sh
+    source "$GWMS_SOURCEDIR"/utils_xml.sh
+    source "$GWMS_SOURCEDIR"/utils_crypto.sh
+    source "$GWMS_SOURCEDIR"/glidein_cleanup.sh
```

Comment:
These do not exist yet. They are packed in the tar add-on. They can be sourced only after you invoke `extract_all_data` (which sources them already)